### PR TITLE
Update ArrayDimFetchToMethodCall to use ObjectType

### PR DESCRIPTION
**Pull request by peterfox**
Follow up to conversation in #5727 @samsonasik 😅

### CODE REVIEW

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Thank you @peterfox

File: rules/Transform/Rector/ArrayDimFetch/ArrayDimFetchToMethodCallRector.php [hunk 1]
```diff
@@ -63,7 +63,7 @@ public function refactor(Node $node): ?MethodCall
         }
 
         foreach ($this->arrayDimFetchToMethodCalls as $arrayDimFetchToMethodCall) {
-            if (! $this->isObjectType($node->var, new ObjectType($arrayDimFetchToMethodCall->getClass()))) {
+            if (! $this->isObjectType($node->var, $arrayDimFetchToMethodCall->getObjectType())) {
```
Comment (samsonasik) [inline rules/Transform/Rector/ArrayDimFetch/ArrayDimFetchToMethodCallRector.php:66]: @peterfox I think that `$arrayDimFetchToMethodCall->getMethod()` should be equal with `$node->dim` value, which `$node->dim->value` value need to be `String_` node, wdyt?

Comment (samsonasik) [reply rules/Transform/Rector/ArrayDimFetch/ArrayDimFetchToMethodCallRector.php:66]: Oh, I see, it used as MethodCall creation, seems ok then 👍

Comment (peterfox) [reply rules/Transform/Rector/ArrayDimFetch/ArrayDimFetchToMethodCallRector.php:66]: @samsonasik I'm not sure I follow. Are you suggesting that the type of the ArrayDimFetch should be analysed to make sure it's a string value and if it is not a string value, to skip refactoring?

Comment (samsonasik) [reply rules/Transform/Rector/ArrayDimFetch/ArrayDimFetchToMethodCallRector.php:66]: nvm, It seems fine as it is, I just missread of how it should transform

**Pull request merged**